### ISSUE TITLE
error app-pouchdb elements with pouchdb-5.4.4

### ISSUE THREAD

**Issue by teoqba** [PolymerElements/app-pouchdb#18]
I have downloaded app-puchdb components through bower. It install pouchdb-5.4.4.
When I import a `app-pouchdb-query` I got an error in JavaScript Console:

```
 Uncaught TypeError: Cannot read property 'Promise' of undefined(anonymous function)  pouchdb.find.js:4493
```

The same application works fine when I revert back to pouchdb-5.3.2

**Comment by raydaly** [comment]
App-pouchdb is an exciting addition for the Polymer Project. However, when writing up a tutorial on how to use app-pouchdb it was discovered that two of the components are not working. They were installed using bower: app-pouchdb-index and app-pouchdb-query. There is no need to detail how to recreate these errors because information in the timeline below.

In investigating and debugging these errors the following timeline became apparent.

- May 18 
  - last major app-pouchdb commit; last commit for bower.json
  - pouchdb v0.5.3.2; pouchdb-find v0.9.0
- Jun 6  
  - pouchdb v5.4.0 which depreciates PouchDB.utils which was used in pouchdb-find
  - pouchdb-find v10.0.0 released (assumed to deal with depricated Pouchdb.utils and more)
- Jun 8 - pouchdb v5.4.1
- Jun 12 - pouchdb v5.4.2
- Jun 13/14 
  - pouchdb-find minor fixes to v0.10.3 (current version)
  - pouchdb v5.4.3
- Jun 16 - pouchdb v5.4.4
- Jul 2  - pouchdb v5.4.5
- Sep  5 - pouchdb v6.0.3 and v6.0.4
- Sep 15 - wrote up this report

This timelime shows that since app-pouchdb was commited on May 18 that a new version of pouchdb (v5.4.0) was introduced with a breaking change that effected pouchdb-find. Thus a new release of pouchdb-find was created (v0.10.0).

Running bower install --save PolymerElements/app-pouchdb results in pouchdb-find#0.9.0 and pouchdb#5.4.5 being install in bower_components. These are not compatible.

So the current bower.json file for app-pouchdb needs to be updated to fix this incompatibility. While that is being done, it is probably worth updating to the latest versions for both and testing with the new versions.

I'm just starting to test app-pouchdb with the latest versions of pouchdb and pouchdb-find. Assuming that is successful, a pull request will be made.

**Comment by e111077** [comment]
@raydaly nice catch. It does seem to resolve the issue. I have a branch open, but I feel like you need the cred for the fix. If you can submit a PR quickly with a minor version bump to both pouchdb and pouchdb-find, I'll be happy to merge it.

**Comment by nolanlawson** [comment]
Yes indeed, I (stupidly) made a breaking change in 5.4.0 to remove `PouchDB.utils`. I didn't consider it a breaking change at the time, because `PouchDB.utils` was always undocumented and unrecommended. Unfortunately it had still found its way into _lots_ of plugins, and so when we released 5.4.0 it broke a ton of dependencies (including many of my own plugins like `pouchdb-find` – stupid me! 😛 ).

The latest versions of both pouchdb and pouchdb-find should work fine together; if not, you can add the [pouchdb-legacy-utils](https://github.com/nolanlawson/pouchdb-legacy-utils) plugin to any project and it will add the missing `.utils` API. Sorry for the headache!

**Comment by raydaly** [comment]
Thanks @e111077. I did a pull request, just hope I did it properly. Have used git for years but first pull request on github.

And thanks @nolanlawson for the comment.
